src/patternfly/components/Nav/nav.scss

```diff
@@ -354,389 +235,275 @@
-      left: var(--pf-c-nav__list-link--after--Left);
-      width: var(--pf-c-nav__list-link--after--Width);
-      height: var(--pf-c-nav__list-link--after--Height);
-      content: "";
-    }
+  &::before {
+    position: absolute;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    height: var(--pf-c-nav__item--before--Height);
+    content: "";
+    background-color: var(--pf-c-nav__item--before--BackgroundColor);
   }
+}
 
-  // Link
-  .pf-c-nav__link {
-    position: relative;
-    display: flex;
-    align-items: baseline;
-    padding:
-      var(--pf-c-nav__list-link--PaddingTop)
-      var(--pf-c-nav__list-link--PaddingRight)
-      var(--pf-c-nav__list-link--PaddingBottom)
-      var(--pf-c-nav__list-link--PaddingLeft);
-    font-weight: var(--pf-c-nav__list-link--FontWeight);
-    color: var(--pf-c-nav__list-link--Color);
-
-
-    // States
-    &.pf-m-hover,
-    &:hover {
-      --pf-c-nav__list-link--Color: var(--pf-c-nav__list-link--hover--Color);
-
-      &::after {
-        background-color: var(--pf-c-nav__list-link--hover--after--BackgroundColor);
-      }
-    }
-
-    &.pf-m-active,
-    &:active {
-      --pf-c-nav__list-link--FontWeight: var(--pf-c-nav__list-link--active--FontWeight);
-      --pf-c-nav__list-link--Color: var(--pf-c-nav__list-link--active--Color);
-
-      &::after {
-        background-color: var(--pf-c-nav__list-link--active--after--BackgroundColor);
-      }
-    }
-
-    &.pf-m-focus,
-    &:focus {
-      --pf-c-nav__list-link--FontWeight: var(--pf-c-nav__list-link--focus--FontWeight);
-      --pf-c-nav__list-link--Color: var(--pf-c-nav__list-link--focus--Color);
-
-      &::after {
-        background-color: var(--pf-c-nav__list-link--focus--after--BackgroundColor);
-      }
-    }
+// Nav link
+.pf-c-nav__link {
+  position: relative;
+  display: flex;
+  align-items: baseline;
+  padding: var(--pf-c-nav__link--PaddingTop) var(--pf-c-nav__link--PaddingRight) var(--pf-c-nav__link--PaddingBottom) var(--pf-c-nav__link--PaddingLeft);
+  font-size: var(--pf-c-nav__link--FontSize);
+  font-weight: var(--pf-c-nav__link--FontWeight);
+  color: var(--pf-c-nav__link--Color);
+  background-color: var(--pf-c-nav__link--BackgroundColor);
+
+  &::before {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    content: "";
+    border-color: var(--pf-c-nav__link--before--BorderColor);
+    border-style: solid;
+    border-width: 0 0 var(--pf-c-nav__link--before--BorderBottomWidth) var(--pf-c-nav__link--before--BorderLeftWidth);
   }
 
   // States
-  .pf-m-current.pf-c-nav__link,
-  .pf-m-current > .pf-c-nav__link {
-    --pf-c-nav__list-link--FontWeight: var(--pf-c-nav__list-link--m-current--FontWeight);
-    --pf-c-nav__list-link--Color: var(--pf-c-nav__list-link--m-current--Color);
-
-    &::after {
-      background-color: var(--pf-c-nav__list-link--m-current--after--BackgroundColor);
-    }
+  &:hover {
+    color: var(--pf-c-nav__link--hover--Color);
```

Comment:
should color and background-color redefine `--pf-c-nav__link--hover` and `--pf-c-nav__link--BackgroundColor` for these states?